src/signature.c

```diff
@@ -82,20 +100,38 @@ bool initialize_signature(void)
 
 	ret = validate_certificate();
 	if (ret) {
-		printf("Failed to verify certificate: %s\n", X509_verify_cert_error_string(ret));
 		if (ret == X509_V_ERR_CERT_NOT_YET_VALID) {
-			/* If we can retrieve an approx. good system time, report out to user */
-			if (stat("/usr/lib/os-release", &statt) != -1) {
-				mod_sec = statt.st_mtim.tv_sec;
-				char timebuf[30];
-				alttime = localtime(&mod_sec);
-				strftime(timebuf, sizeof(timebuf), "%F", alttime);
-				printf("System clock should be at least %s\n", timebuf);
+			time_t currtime;
```

Comment:
Sure, then I think the bad cert case is really still deciding if the system time is good or not first, and if it's good and still fails, the cert must be bad so quit.If the system time seems weird then we try to fix it and check again like normal. I'll make some changes and push again.